apps/web/src/services/asset-inventory/components/CollectorDataModal.vue

```diff
@@ -122,13 +122,14 @@ const fetchSecrets = async (provider: string, serviceAccounts: string[]) => {
     }
 };
 
-watch([() => collectorDataModalState.selectedCollector, () => collectorDataModalState.visible], async ([selectedCollector, visible]) => {
-    if (!selectedCollector || !visible) {
+watch([() => selectedCollectorData.value, () => collectorDataModalState.visible], async ([_selectedCollectorData, visible]) => {
+    if (!visible) {
         collectorDataModalStore.$reset();
         return;
     }
-    await fetchSecrets(selectedCollector.provider, state.serviceAccountsFilter);
-    await collectorDataModalStore.getJobs(selectedCollector.collector_id);
+    if (!_selectedCollectorData) return;
+    await fetchSecrets(_selectedCollectorData.provider, state.serviceAccountsFilter);
+    await collectorDataModalStore.getJobs(_selectedCollectorData.collector_id);
 }, { immediate: true });
```

Comment:
Question: Is `collectorDataModalStore.getJobs` action really necessary, isn't it already getting job as `useCollectorGetQuery` above?